llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -3440,14 +3531,6 @@ Value *RISCVTargetLowering::emitMaskedAtomicCmpXchgIntrinsic(
   return Result;
 }
 
-bool RISCVTargetLowering::canLowerPointerTypeCmpXchg(
-    const llvm::DataLayout &DL, llvm::AtomicCmpXchgInst *AI) const {
-  if (Subtarget.hasStdExtA() && Subtarget.hasCheri() &&
```

Comment:
Does deleting the A check not break non-A codegen?

Reply:
As you can see in llvm/test/CodeGen/CHERI-Generic/RISCV64/cmpxchg-cap-ptr.ll, removing a results in libcalls, so it does not seem to be required.
If I recall correctly, this function is only used to determine whether the IR can contain cmpxchg instructions with pointer types instead of just integers (since non-RISC-V/non-MIPS backends expect integer operands).